Add the feature to read eui64 from flash area

pulin1103 opened the pull request:
http://192.168.48.49:8080/browse/UTCSH-381
Please tell me if there are something incorrect.

andriy-bilynskyy commented on drivers/ieee802154/Kconfig.b9x:
```diff
@@ -36,14 +36,28 @@ config IEEE802154_B9X_RANDOM_MAC
 	help
 	  Generate a random MAC address dynamically.
 
+config IEEE802154_B9X_FLASH_MAC
```
Previously we had just two options for IEEE802154 MAC: random and hardcoded.
So we used boolean option to select address type - IEEE802154_B9X_RANDOM_MAC.
Now we are going to have 3 address types: random, hardcoded and vendor flash partition. Only one type is valid.
In KConfig there is _choice_ construction for such stuffs.
See: [tlsr9/ble/vendor/controller/Kconfig in hal_telink](https://github.com/telink-semi/hal_telink/blob/develop/tlsr9/ble/vendor/controller/Kconfig)
`choice B9X_BLE_CTRL_MAC_TYPE`
as example. In that case only one option can be selected. In that case you avoid strange constructions like
`if ! IEEE802154_B9X_RANDOM_MAC && ! IEEE802154_B9X_FLASH_MAC`
which implicitly provides prioritization.

andriy-bilynskyy commented on drivers/ieee802154/ieee802154_b9x.c:
```diff
@@ -444,6 +446,9 @@ static ALWAYS_INLINE uint8_t *b9x_get_mac(const struct device *dev)
 	 */
 	b9x->mac_addr[0] = (b9x->mac_addr[0] & ~0x01) | 0x02;
+#elif defined(CONFIG_IEEE802154_B9X_FLASH_MAC)
+	uintptr_t addr_flash_mac = CONFIG_FLASH_BASE_ADDRESS + CONFIG_IEEE802154_B9X_FLASH_MAC_OFFSET;
```
We should avoid using hal drivers them are not thread safe.
We should use Zephyr flash driver. vendor_partition partition is already implemented.
See https://github.com/telink-semi/hal_telink/blob/develop/tlsr9/ble/vendor/controller/b9x_bt_flash.c
here is the same stuff as you implementing for Thred just for BLE.

Commit:
drivers: ieee802154: b9x: Add the feature to read eui64 from flash area

- Read IEEE802154 MAC adress from flash: CONFIG_IEEE802154_B9X_MAC_FLASH
- Get random IEEE802154 MAC adress: CONFIG_IEEE802154_B9X_MAC_RANDOM
- Use precompiled IEEE802154 MAC adress: CONFIG_IEEE802154_B9X_MAC_STATIC

Signed-off-by: Zhenghuan Zhang <zhenghuan.zhang@telink-semi.com>
